front50-core/src/main/groovy/com/netflix/spinnaker/front50/config/CommonServiceConfig.java

```diff
@@ -1,47 +0,0 @@
-/*
- * Copyright 2020 Adevinta
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.netflix.spinnaker.front50.config;
-
-import com.netflix.spinnaker.fiat.shared.FiatClientConfigurationProperties;
-import com.netflix.spinnaker.fiat.shared.FiatPermissionEvaluator;
-import com.netflix.spinnaker.fiat.shared.FiatService;
-import com.netflix.spinnaker.front50.ServiceAccountsService;
-import com.netflix.spinnaker.front50.model.serviceaccount.ServiceAccountDAO;
-import java.util.Optional;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-
-@Configuration
-public class CommonServiceConfig {
-
-  @Bean
-  @ConditionalOnBean(ServiceAccountDAO.class)
```

Comment:
`@ConditionalOnBean` stuff is pretty finicky outside the AutoConfiguration lifecycle and we are relying on `@ComponentScan` to load all the providers.

I could get behind switching this to `@ConditionalOnExpression("not ${spinnaker.redis.enabled:false}")` (that SpEL is likely maybe wrong)

Reply:
I guess this would mess up someone who is trying to have redis and another provider enabled for migration buutttt I'm still okay with that personally